### DISCUSSION TITLE
If there is a way on filament 3 site to show custom lines on any resource page ?

### DISCUSSION THREAD

**Discussion by PetroGromovo** [filamentphp/filament#14114] [Help]
### Package

Other

### Package Version

3.2

### How can we help you?

In filament 2 site I had access to template of resource pages in resources/views/vendor/filament/components/header/index.blade.php file
and editing this file like :

```
@props([
'actions' => null,
'heading',
'subheading' => null
])

<header {{ $attributes->class(['filament-header space-y-2 items-start justify-between sm:flex sm:space-y-0 sm:space-x-4  sm:rtl:space-x-reverse sm:py-4']) }}>
    <div>
        <x-filament::header.heading>
            <div class="flex items-center">
                <div class="mr-2 w-16"><img src="/images/app_icon.png" alt="App icon"></div>
                <div>{{ getAppTitleVersion() }} : {{ $heading }}</div>
            </div>
        </x-filament::header.heading>

        @if ( function_exists('getAppSubheading'))
        <x-filament::header.subheading class="mt-1">{{ getAppSubheading() }}</x-filament::header.subheading>
        @else

        @if ($subheading)
        <x-filament::header.subheading class="mt-1">{{ $subheading }}</x-filament::header.subheading>
        @endif

        @endif
    </div>


    <x-filament::pages.actions :actions="$actions" class="shrink-0" />
</header>


```
and using custom functions getAppTitleVersion() and getAppSubheading() I was able to show some custom text relative to current user and opened page.

If there is such a functionality in filament 3?

```
"laravel/framework": "^11.19",
"filament/filament": "^3.2-stable",

```
Thanks in advance!

**Comment by leandrocfe** [comment]
Try render hooks

https://filamentphp.com/docs/3.x/support/render-hooks/#overview

**Reply by PetroGromovo** [reply]
Looking at provided link I found  method RESOURCE_PAGES_LIST_RECORDS_TABLE_BEFORE usefull for resource list, but I did not find any similar method for form - i need to show
additive info just under title of form editor. Has it any ?
